### PR TITLE
Fix issue in CFG computation with generic record pattern

### PR DESCRIPTION
**Pull request by romainbrenguier**

### CODE REVIEW

File: java-frontend/src/test/java/org/sonar/java/cfg/CFGTest.java [hunk 1]
```diff
@@ -3149,6 +3151,37 @@ void throw_statement_within_try_catch() {
 
   }
 
+  @Test
+  void generic_record_pattern() {
+    CFG cfg = buildCFG("""
+        private static boolean testGen3(Object o) throws Throwable {
+          return o instanceof GenRecord1<?, ?>(Integer i, var s) && i.intValue() == 3 && s.length() == 0;
```
Comment (dorian-burihabwa-sonarsource) [inline java-frontend/src/test/java/org/sonar/java/cfg/CFGTest.java]: I am trying some corner cases here but we should probably test with a bounded wild card

Comment (dorian-burihabwa-sonarsource) [reply java-frontend/src/test/java/org/sonar/java/cfg/CFGTest.java]: `return o instanceof GenRecord1<?, ? extends CharSequence>(Integer i, var s) && i.intValue() == 3 && s.length() == 0;`
